opentelemetry-api/src/opentelemetry/trace/span.py

```diff
@@ -118,6 +118,23 @@ def add_event(
         timestamp if the `timestamp` argument is omitted.
         """
 
+    @abc.abstractmethod
+    def add_link(
+        self,
+        context: "SpanContext",
+        attributes: types.Attributes = None,
```

Comment:
This should accept [`Link`](https://github.com/open-telemetry/opentelemetry-python/blob/975733c71473cddddd0859c6fcbd2b02405f7e12/opentelemetry-api/src/opentelemetry/trace/__init__.py#L133) which is already part of the trace api.

Reply:
I'm following the otel-specification defined [here](https://opentelemetry.io/docs/specs/otel/trace/api/#link) which states AddLink() should accept span context and (optionally) attributes. The [Cpp](https://github.com/open-telemetry/opentelemetry-specification/pull/3678) and [Java](https://github.com/open-telemetry/opentelemetry-java/pull/6084) implementation also followed the same API design.

Reply:
> either as individual parameters or as an immutable object encapsulating them, whichever is most appropriate for the language

This is the reason I suggested but give other implementation have followed two args approach this is fine.